src/main/java/com/iota/iri/network/Node.java

```diff
@@ -410,7 +410,7 @@ public void replyToRequest(Hash requestedHash, Neighbor neighbor) {
         if (requestedHash.equals(Hash.NULL_HASH)) {
             //Random Tip Request
             try {
-                if (transactionRequester.numberOfTransactionsToRequest() > 0 && rnd.nextDouble() < P_REPLY_RANDOM_TIP) {
+                if (rnd.nextDouble() < P_REPLY_RANDOM_TIP) {
```

Comment:
This change requires a little explanation

Reply:
well as far as i know the transactionRequester is responsible for fetching tx which are unknown to the node. this piece of code just needs to reply to the neighbor with a random tip (from tipsViewModel). so i do not see the need of a positive numberOfTransactionsToRequest when replying.